### PR TITLE
[feat] Improved validation ID and messages

### LINKED ISSUE #145: cracked cluster because of the conn id
**Description:**
Cluster fails because of the id of the connections like `go/clone`, `ru-https:/`, etc. for types of conn. as `git`, `gcs`, and maybe for others.
MicrosoftTeams-image (3)
ru-https

### LINKED ISSUE #147: make changes to 'error' descriptions
**Description:**
Here is a list of proposed changes to the error description.

**List:**
1. Should be "Error:  **maximum** number of replicas parameter must not be less than **minimum** number of replicas parameter"
MicrosoftTeams-image (4)

1. If a connection already exists I got the error:
MicrosoftTeams-image (5)
It should be "Error: connections.odahuflow.odahu.org "`connection name`" already exists".

### PR DESCRIPTION
**Pull request by mcdoker18**
This closes #147 
This closes #145

### CODE REVIEW

File: packages/operator/pkg/validation/validation.go [hunk 1]
```diff
@@ -41,3 +44,19 @@ func ValidateExistsInRepository(name string, repository connection.Repository) e
...
+
+// Id restrictions:
+//  * contain at most 63 characters
+//  * contain only lowercase alphanumeric characters or ‘-’
+//  * start with an alphanumeric character
+//  * end with an alphanumeric character
+func ValidateID(id string) error {
+	if idRegex.MatchString(id) && !strings.HasPrefix(id, "-") && !strings.HasSuffix(id, "-") {
+		return nil
+	}
```
Comment (karbyshevds) [inline packages/operator/pkg/validation/validation.go]: It's not obvious, can't we declare valid regexp that will cover all naming conventions and just check that it's matching?

Comment (mcdoker18) [reply packages/operator/pkg/validation/validation.go]: I am appreciate any help with the regex expression. :)

Comment (maxkochubey) [reply packages/operator/pkg/validation/validation.go]: You can use that yesterday regex :)

Comment (mcdoker18) [reply packages/operator/pkg/validation/validation.go]: @karbyshevds Thanks for help, fixed

**Pull request merged**